packages/core/saved-objects/core-saved-objects-server-internal/src/routes/create.ts

```diff
@@ -56,6 +57,13 @@ export const registerCreateRoute = (
       const usageStatsClient = coreUsageData.getClient();
       usageStatsClient.incrementSavedObjectsCreate({ request: req }).catch(() => {});
 
+      const { savedObjects } = await context.core;
+
+      const fullType = savedObjects.typeRegistry.getType(type);
+      if (!fullType?.hidden && fullType?.hiddenFromHttpApis) {
+        throw SavedObjectsErrorHelpers.createUnsupportedTypeError(type);
+      }
```

Comment:
NIT/optional: Same for non-bulk APIs, even if it's only converting 3 lines to 1, I would create a `throwIfTypeNotVisibleByAPI(type: string, registry: ISavedObjectTypeRegistry)` to factorize

Reply:
Same as https://github.com/elastic/kibana/pull/149166#discussion_r1084181213

Reply:
Done [`a48dfdd` (#149166)](https://github.com/elastic/kibana/pull/149166/commits/a48dfdd78c341430daf4578d16f43f23d1e81696)